lua/roslyn/init.lua

```diff
@@ -12,21 +14,58 @@ local function get_mason_installation()
         or mason_installation
 end
 
----@type string?
-local _pipe_name = nil
+-- Assigns the default capabilities from cmp if installed, and the capabilities from neovim
+---@param roslyn_config? RoslynNvimConfig
+local function get_default_capabilities(roslyn_config)
+    local ok, cmp_nvim_lsp = pcall(require, "cmp_nvim_lsp")
+    local capabilities = ok
+            and vim.tbl_deep_extend(
+                "force",
+                vim.lsp.protocol.make_client_capabilities(),
+                cmp_nvim_lsp.default_capabilities()
+            )
+        or vim.lsp.protocol.make_client_capabilities()
+
+    -- This actually tells the server that the client can do filewatching.
+    -- We will then later just not watch any files. This is because the server
+    -- will fallback to its own filewatching which is super slow.
+
+    -- Default value is true, so the user needs to explicitly pass `false` for this to happen
+    -- `not filewatching` evaluates to true if the user don't provide a value for this
+    if roslyn_config and roslyn_config.filewatching == false then
+        capabilities = vim.tbl_deep_extend("force", capabilities, {
+            workspace = {
+                didChangeWatchedFiles = {
+                    dynamicRegistration = true,
+                },
+            },
+        })
+    end
+
+    -- HACK: Roslyn requires the dynamicRegistration to be set to support diagnostics for some reason
+    return vim.tbl_deep_extend("force", capabilities, {
+        textDocument = {
+            diagnostic = {
+                dynamicRegistration = true,
+            },
+        },
+    })
+end
 
 ---@type table<string, string>
 ---Key is solution directory, and value is sln target
 local known_solutions = {}
 
 ---@param pipe string
 ---@param target string
----@param config vim.lsp.ClientConfig
+---@param config? vim.lsp.ClientConfig
```

Comment:
This don't need to be optional anymore I think

Reply:
yeah, sorry, rebased and amended the previous commit 😅 see https://github.com/seblj/roslyn.nvim/pull/25/commits/69a681b0897a5cad400cd90a9d101c6d15135b96